src/schema/validator.js

```diff
@@ -163,7 +166,7 @@ function getManifestVersionsRange(validatorOptions) {
     );
   }
 
-  return { minimum, maximum };
+  return { minimum, maximum, addon };
```

Comment:
I renamed it to currentAddon (just to make it clear it is the version of the current addon being validated, and not a current manifest version option set on the linter, like minimum and maximum are).

Let me know if that is a reasonable name from this property also from your perspective.

Reply:
Ok so that's my fault actually. I should have suggested something else: I don't like that we return the addon manifest version in something called "range". This is no longer a range anymore since it has more than min/max.

I tried to move the logic for `currentAddon` where it is actually used and the test suite still passes:

```diff
diff --git a/src/schema/validator.js b/src/schema/validator.js
index 6c700742..57f0ef25 100644
--- a/src/schema/validator.js
+++ b/src/schema/validator.js
@@ -140,8 +140,7 @@ function filterErrors(
 }
 
 function getManifestVersionsRange(validatorOptions) {
-  const { minManifestVersion, maxManifestVersion, addonManifestVersion } =
-    validatorOptions;
+  const { minManifestVersion, maxManifestVersion } = validatorOptions;
 
   const minimum =
     minManifestVersion == null
@@ -164,10 +163,7 @@ function getManifestVersionsRange(validatorOptions) {
     );
   }
 
-  const currentAddon =
-    addonManifestVersion == null ? minimum : addonManifestVersion;
-
-  return { minimum, maximum, currentAddon };
+  return { minimum, maximum };
 }
 
 export class SchemaValidator {
@@ -413,8 +409,11 @@ export class SchemaValidator {
   }
 
   _compileAddonValidator(validator) {
-    const { minimum, currentAddon, maximum } =
-      this.allowedManifestVersionsRange;
+    const { minimum, maximum } = this.allowedManifestVersionsRange;
+    const { addonManifestVersion } = this._options;
+
+    const manifestVersion =
+      addonManifestVersion == null ? minimum : addonManifestVersion;
 
     const replacer = (key, value) => {
       if (Array.isArray(value)) {
@@ -427,7 +426,7 @@ export class SchemaValidator {
             includeItem =
               item.min_manifest_version >= minimum &&
               item.min_manifest_version <= maximum &&
-              item.min_manifest_version <= currentAddon;
+              item.min_manifest_version <= manifestVersion;
           }
           if (
             item?.max_manifest_version &&
@@ -436,7 +435,7 @@ export class SchemaValidator {
             includeItem =
               item.max_manifest_version >= minimum &&
               item.max_manifest_version <= maximum &&
-              item.max_manifest_version >= currentAddon;
+              item.max_manifest_version >= manifestVersion;
           }
 
           return includeItem;
```

WDYT?

Reply:
@willdurand ah, now I see what you meant, I applied your diff and made some more small tweaks on top of it (basically moved it into a Validator class getter, see https://github.com/mozilla/addons-linter/pull/4397/commits/3c2a0a7c72e0d92f29da3c110b9e05f5f64a77f9), how that looks to you?